### PR TITLE
turn hello.ml into a more convenient test

### PR DESCRIPTION
**Pull request by gasche**
Now you can run

    make compiler-test

from the root of the repository to run hello.ml, and compare its
output to a version-controlled reference output.

If the output changes for good reasons, you can update the reference
file with

    make compiler-test-promote

Note: it may sound unpleasant to version-control program-generated
outputs, but in fact (as long as those outputs are kept
human-readable) this is an excellent idea for testing: it means that
any change to the testing output will be part of the commits/diffs,
making it easier for reviewers to understand the impact of a change.

### CODE REVIEW

File: miniml/compiler/test/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+If you have already run 'make ocamlrun' at the root, you may use the following:
+
+make OCAMLRUN=../../../_boot/byterun/ocamlrun
+
+
+endef
+
+
+COMPILE_CMD=GUILE_WARN_DEPRECATED=detailed guile ../compile.scm
```
Comment (Ekdohibs) [inline miniml/compiler/test/Makefile:20]: What is `detailed`?

**Review by Ekdohibs — COMMENTED** [review]
Seems good! Could you please also add `*.byte` and `*.output` to `.gitignore`?

**Comment by Ekdohibs** [comment]
I don't have any objection to committing program-generated outputs, as long as their purpose is testing and not building the source. Besides, I'd count test outputs as source code.

**Comment by gasche** [comment]
> Could you please also add *.byte and *.output to .gitignore?

Done.

File: miniml/compiler/test/Makefile (hunk 1 above)
Comment (Ekdohibs) [reply miniml/compiler/test/Makefile:20]: Oh, I hadn't read the `GUILE_WARN_DEPRECATED` part and I thought `detailed` was a command.

**Pull request merged**